core/capabilities/ccip/common/addresscodec.go

```diff
@@ -4,46 +4,25 @@ import (
 	"fmt"
 
 	chainsel "github.com/smartcontractkit/chain-selectors"
-
 	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
 )
 
-// ChainSpecificAddressCodec is an interface that defines the methods for encoding and decoding addresses
-type ChainSpecificAddressCodec interface {
-	AddressBytesToString([]byte) (string, error)
-	AddressStringToBytes(string) ([]byte, error)
-}
-
-// AddressCodec is a struct that holds the chain specific address codecs
-type AddressCodec struct {
-	EVMAddressCodec    ChainSpecificAddressCodec
-	SolanaAddressCodec ChainSpecificAddressCodec
-}
-
-// AddressCodecParams is a struct that holds the parameters for creating a AddressCodec
-type AddressCodecParams struct {
-	evmAddressCodec    ChainSpecificAddressCodec
-	solanaAddressCodec ChainSpecificAddressCodec
-}
-
-// NewAddressCodecParams is a constructor for AddressCodecParams
-func NewAddressCodecParams(evmAddressCodec ChainSpecificAddressCodec, solanaAddressCodec ChainSpecificAddressCodec) AddressCodecParams {
-	return AddressCodecParams{
-		evmAddressCodec:    evmAddressCodec,
-		solanaAddressCodec: solanaAddressCodec,
-	}
+// RealAddressCodec is a struct that holds the chain specific address codecs
+type RealAddressCodec struct {
```

Comment:
Is this prefaced with `Real` because there's a `Fake` somewhere? Or is this basically intended to be a singleton w/ all the relevant codecs?

Reply:
There's an interface named `AddressCodec`.

We had a discussion about this in September.
```
1. func NewFoo() *fooStruct

2. func NewFoo() *FooStruct

3. func NewFoo() FooInterface
```

If we *really* need an interface (for mocking) then I would slightly modify the code so that this struct is private and we return the interface from `NewAddressCodec`.

In this case, I would try to use the real codec in tests and have a single `AddressCodec` which is also the actual implementation.

In both cases, I wouldn't name something `Real` in a public interface. Someone using the `common` package should only be allowed to use the "Real" thing.

Some styleguide references:
* https://go.dev/doc/effective_go#generality
* https://google.github.io/styleguide/go/decisions.html#interfaces